crates/vm/src/arch/integration_api.rs

```diff
@@ -341,6 +458,29 @@ where
     }
 }
 
+/// RecordArena implementation for [DenseRecordArena], with [AdapterCoreLayout]
+/// `A` is the adapter record type and `C` is the core record type
```

Comment:
```suggestion
/// `A` is the adapter RecordMut type and `C` is the core RecordMut type
```

This is a mistake from my side, missed it in the above comment.